lib/fs/basicfs.go

```diff
@@ -53,34 +52,17 @@ func newBasicFilesystem(root string) *BasicFilesystem {
 		}
 	}
 
-	rootSymlinkEvaluated, err := filepath.EvalSymlinks(root)
-	if err != nil {
-		rootSymlinkEvaluated = root
-	}
-
-	return &BasicFilesystem{
-		root:                 adjustRoot(root),
-		rootSymlinkEvaluated: adjustRoot(rootSymlinkEvaluated),
-	}
-}
-
-func adjustRoot(root string) string {
 	// Attempt to enable long filename support on Windows. We may still not
 	// have an absolute path here if the previous steps failed.
 	if runtime.GOOS == "windows" {
-		if filepath.IsAbs(root) && !strings.HasPrefix(root, `\\`) {
-			root = `\\?\` + root
-		}
-		return root
-	}
-
-	// If we're not on Windows, we want the path to end with a slash to
-	// penetrate symlinks. On Windows, paths must not end with a slash.
-	if root[len(root)-1] != filepath.Separator {
+		root = longFilenameSupport(root)
+	} else if root[len(root)-1] != filepath.Separator {
```

Comment:
I'd do a real strings.HasSuffix here instead, avoids crashing on the empty string etc

Reply:
Done. I also adapted the windows fs creation test for unixes to check this and other basic scenarios. Apparently the tilde and absolutification meddling above already changed "" to "/", so it wouldn't panic even before.